src/main.py

```diff
@@ -22,23 +25,22 @@ def reformat_varbinary_columns(df, varbinary_columns):
 
 
 def fetch_dune_data():
-    response = requests.get(API_URL, headers=HEADERS, timeout=10)
-    if response.status_code == 200:
-        data = response.json()
-        result = data["result"]
-        metadata = result["metadata"]
-        dtypes, varbinary_columns = {}, []
-        for name, d_type in zip(metadata["column_names"], metadata["column_types"]):
-            dtypes[name] = DUNE_TO_PG[d_type]
-            if d_type == "varbinary":
-                varbinary_columns.append(name)
-        df = pd.DataFrame(result["rows"])
-        # escape bytes
-        df = reformat_varbinary_columns(df, varbinary_columns)
-        return df, dtypes
-
-    print(f"Error fetching data: {response.status_code}")
-    return None
+    result = (
+        DuneClient(DUNE_API_KEY, performance=QUERY_ENGINE)
+        .run_query(query=QueryBase(QUERY_ID), ping_frequency=POLL_FREQUENCY)
+        .result
+    )
+
+    metadata = result.metadata
+    dtypes, varbinary_columns = {}, []
+    for name, d_type in zip(metadata.column_names, metadata.column_types):
+        dtypes[name] = DUNE_TO_PG[d_type]
+        if d_type == "varbinary":
+            varbinary_columns.append(name)
+    df = pd.DataFrame(result.rows)
+    # escape bytes
+    df = reformat_varbinary_columns(df, varbinary_columns)
+    return df, dtypes
 
 
 def save_to_postgres(df, dtypes):
```

Comment:
nitpick: this function could use some error handling and logging.

Reply:
Apropos logging - line 50 is a print().

Reply:
Ya great point. We may have to build out this entire functionality into a class anyway. Lets call it error handling for now and see if its necessary to take it any further.

Reply:
- Error Handling captured by #6
- Logging captured by #7